zng/resolver/marshal.go

```diff
@@ -104,6 +122,98 @@ func fieldName(f reflect.StructField) string {
 	return f.Name
 }
 
+func typeSimple(name, path string) string {
+	return name
+}
+
+func typePackage(name, path string) string {
+	a := strings.Split(path, "/")
+	return fmt.Sprintf("%s.%s", a[len(a)-1], name)
+}
+
+func typeFull(name, path string) string {
+	return fmt.Sprintf("%s.%s", path, name)
+}
+
+const (
+	TypeNone = iota
+	TypeSimple
+	TypePackage
+	TypeFull
+)
```

Comment:
I think there should be a type for these like `type Style int` or `type TypeStyle int`. Also, they read like types because of the Type prefix, so maybe change that to Style or TypeStyle.